core/sds.c

```diff
@@ -1454,6 +1454,13 @@ _sds_upload_add_headers(struct oio_sds_ul_s *ul, struct http_put_dest_s *dest)
 			"%s", escaped);
 	g_free (escaped);
 
+	const gchar *full_path = oio_url_get(ul->dst->url, OIOURL_WHOLE);
```

Comment:
Notice that `oio_url_get(ul->dst->url, OIOURL_WHOLE);` prepends namespace name and appends object version. I'm not sure that's what you want. You should add a note somewhere telling that version is not needed (or append it everywhere if it is).